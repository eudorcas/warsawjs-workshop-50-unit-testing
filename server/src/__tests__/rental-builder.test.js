const Rental = require('../rentals/rental');
const RentalBuilder = require('../rentals/rental-builder');

test('should return rental', () => {
  const rental = new RentalBuilder()
    .rentBy(1)
    .selectCar(22)
    .depositAmount(1000)
    .inState(Rental.RESERVED)
    .build();

  expect(rental).toMatchSnapshot();
});

test('should return rental paid', () => {
  const rental = new RentalBuilder().buildPaid();

  expect(rental.getState()).toMatchInlineSnapshot(`"DEPOSIT_PAID"`);
});
