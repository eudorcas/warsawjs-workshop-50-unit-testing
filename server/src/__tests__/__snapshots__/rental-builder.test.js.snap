// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return rental 1`] = `
Rental {
  "car_id": 22,
  "client_id": 1,
  "deposit": 1000,
  "rental_id": 1,
  "state": "RESERVED",
}
`;
